### ISSUE TITLE
Feature request: Dynamically change minNights

### ISSUE THREAD

**Issue by Maxximus007** [krystalcampioni/vue-hotel-datepicker#97]
Currently, if there is a minNights > 1, and there are 2 blocked periods that are not consecutive but have less than the minNights available, that specific available period is not selectable.

So, if there is an availability of 3 nights between two blocked periods, but required nights is 5, I'd like to auto-lower the required nights to 3 for that period - and make it possible to select an end date.

That would help to get a better fill rate for properties where minNights is enabled.

I do this currently server-side for the visible calendar (lowering minNight), but it is valid for all displayed dates.

Possible approaches: 
- Calculate nights between blocked periods, and allow end-date selection even if available nights is less than minNights.
- change minNights to an array that will allow different minNights in different periods (eg. from: yymmdd until yymmdd -> minNights 5, etc) .

This deserves a request on it's own, as @amulchinock pointed out.

**Comment by amulchinock** [comment]
@krystalcampioni Label as a feature request, pretty please? :)

**Comment by matiasperrone** [comment]
So... let me understand.
Do you want to ignore the minNights sometimes?

Do you want to add this feature on v3? maybe 3.1?
I do need a more detailed example or examples of the logic. I am not following the logic here... seems that you need not to set the minNights option.

Do you want not to restrict sometimes, maybe with a function?

The array feature seems interesting, thought.

**Comment by Maxximus007** [comment]
Okay let me try to explain.
As an example we have set minNights at 5. This guarantees that bookings will have at least 5 nights. But it's quite possible we have several non-connecting bookings with less than 5 nights in between. 
These periods become unbookable, because you are able to select the startdate but can't select an enddate that adheres to the minNight restriction.
 
Let's say we have 2 booked periods, one ending on day 20, and the next starts at 24. If we have set a minNights of 5, this open period is unbookable since the minNights will never be met.

In that case you'd want to lift or lower the general minNights restriction, so booking for a period shorter than minNights is possible.

If the minNights were set in an array, this could be done server side, and no calculations have to be done client side.
